app/services/role_checker.rb

```diff
@@ -0,0 +1,48 @@
+class RoleChecker
```

Comment:
Reading this again, it strikes me that this is a `ProjectRoleChecker` and clarifying the name helps when reading the code.

Reply:
there are no other types of roles, but ok